### PR TITLE
[jit] Add fractional_max_pool2d to standard lib

### PR DESCRIPTION
**Pull request by driazati**

### CODE REVIEW

File: torch/nn/functional.py [hunk 1]
```diff
@@ -337,18 +338,36 @@ def fractional_max_pool2d(input, kernel_size, output_size=None,
...
+        _output_size = [int(input.size(2) * _output_ratio[0]),
+                        int(input.size(3) * _output_ratio[1])]
     else:
         _output_size = torch.jit._unwrap_optional(output_size)
 
     if _random_samples is None:
-        _random_samples = input.new(input.size(0), input.size(1), 2).uniform_()
-    ret = torch._C._nn.fractional_max_pool2d(input, kernel_size, _output_size, _random_samples)
-    return ret if return_indices else ret[0]
+        _random_samples = torch.rand(input.size(0), input.size(1), 2, dtype=input.dtype)
```
Comment (driazati) [inline torch/nn/functional.py]: We don't have support for `new`, but `rand` pulls from the same uniform distribution so this change should be equivalent

Comment (eellison) [reply torch/nn/functional.py]: Do you also need to set the device as well ?  `By default, the returned Tensor has the same torch.dtype and torch.device as this tensor. `

File: torch/nn/modules/pooling.py [hunk 2]
```diff
@@ -675,12 +678,12 @@ def __init__(self, kernel_size, output_size=None, output_ratio=None,
                 raise ValueError("output_ratio must be between 0 and 1 (got {})"
                                  .format(output_ratio))
 
+    @weak_script_method
     def forward(self, input):
-        samples = None if self._random_samples is None else self._random_samples
```
Comment (eellison) [inline torch/nn/modules/pooling.py]: 🤔

**Review by eellison — APPROVED** [review]
LGTM

**Pull request closed without merging**